### DISCUSSION TITLE
Why webdriverio releases updated version that requires migration of entire project

### DISCUSSION THREAD

**Discussion by praveendvd** [webdriverio/codemod#27] [General]
Hi Team,

Just wondering why there is a need for migration of existing projects into newer versions for each version release. Does this create a maintenance headache?

**Comment by christian-bromann** [comment]
I believe all major version releases require some sort of migration given potential breaking changes. Sometimes this doesn't affect all users, e.g. with the v6 -> v7 update which only was breaking for TypeScript users. We plan such breaking changes carefully to ensure the project can continue to grow while at the same time keeping the burden of updating the project reasonable. In addition to that we have provided [codemods](https://github.com/webdriverio/codemod) that helps with the migration.

**Reply by praveendvd** [reply]
Yes but I read webdriverio releases major versions every year that's a really critical release frequency for breaking changes . Thank you for the great support

**Reply by christian-bromann** [reply]
Yes, we plan to have such releases once a year but that doesn't mean you have to rewrite your code base every time such major release happen. Our codemods can help you automate this process almost in a full automated fashion.
